### PR TITLE
Feat: Add New Demo accounts 

### PR DESCRIPTION
**Pull request by pantharshit007**
Issue #37 

Adding a new account for demo users to increase user interactivity and improve UX.

image
image

Added Vercel.json file for routing compatibility with Vercel.

### CODE REVIEW

**Comment by yesahem** [comment]
Doesn't looks good to me @pantharshit007 
what's you call @geekyharsh05

**Comment by geekyharsh05** [comment]
@pantharshit007 It will be better if you provide an Option below the signup button
Login as guest.

**Comment by pantharshit007** [comment]
Well the modal is closable but alright good call and about what @yesahem  mentioned not allowing demo user to create Blog for that we can simply restrict CUD calls from that particular email and prompt user to create new account
Hows that sound @geekyharsh05

**Comment by geekyharsh05** [comment]
@pantharshit007 You want the guest user to perform CRUD operations?

**Comment by pantharshit007** [comment]
> @pantharshit007 You want the guest user to perform CRUD operations?

Initially I thought
But after what @yesahem  said I drop the idea and limit the demo user to only read operation

**Comment by geekyharsh05** [comment]
@pantharshit007 Make the guest user as  read only,  if he/she wants to create a post then there should be a prompt `create an account to continue`.

**Comment by pantharshit007** [comment]
Yeah thats what I am gona do now

**Comment by pantharshit007** [comment]
SIGN IN 
image

SIGN UP
image

I could not check for any error in the backend part due to some errors setting up Prisma, but it should work. You can check that locally.

Added restriction so that user wouldn't be able to edit update or delete any blog
@yesahem @geekyharsh05

File: backend/src/controllers/blog.controller.ts [hunk 1]
```diff
@@ -47,6 +54,13 @@ export const updateBlogHandler = factory.createHandlers(async (c) => {
   const userid = c.var.userid; // Get the user ID from the context
   const updatedBody: UpdateBlogSchema = await c.req.json();
 
+  // demo account restriction
+  const user = c.get("user");
```
Comment (pantharshit007) [inline backend/src/controllers/blog.controller.ts:58]: this is the part i am talking about @yesahem

**Comment by geekyharsh05** [comment]
@pantharshit007 Looks Fine for now, will merge it after testing locally.

**Comment by geekyharsh05** [comment]
@pantharshit007 What are the issues you are facing in setting up prisma?

**Comment by pantharshit007** [comment]
@geekyharsh05 I was running prisma in docker and when I made a db call I get a prisma initialisation error maybe because I am not using accelerate url or something

But after that my docker stopped working so yeah I wasn't able to cross check the changes but they are minor changes anyway like adding email in token payload and grabbing it again so there shouldn't be any issues.

**Comment by geekyharsh05** [comment]
@pantharshit007 Go to neon.tech spin up a new postgres database, then afterwards go to prisma accelerate generate an accelerate url. Then create a .env in the backend folder with these three Variables.
```bash
DIRECT_DATABASE_URL=Postgres db url of neon

DATABASE_URL=accelerate url

JWT_SECRET=your secret
```

**Comment by pantharshit007** [comment]
> @pantharshit007 Go to neon.tech spin up a new postgres database, then afterwards go to prisma accelerate generate an accelerate url. Then create a .env in the backend folder with these three Variables.
> 
> ```shell
> DIRECT_DATABASE_URL=Postgres db url of neon
> 
> DATABASE_URL=accelerate url
> 
> JWT_SECRET=your secret
> ```

I know all of that it's just that I don't wana do all that for a simple change 😗

**Comment by geekyharsh05** [comment]
> > @pantharshit007 Go to neon.tech spin up a new postgres database, then afterwards go to prisma accelerate generate an accelerate url. Then create a .env in the backend folder with these three Variables.
> > 
> > ```shell
> > DIRECT_DATABASE_URL=Postgres db url of neon
> > 
> > DATABASE_URL=accelerate url
> > 
> > JWT_SECRET=your secret
> > ```
> 
> I know all of that it's just that I don't wana do all that for a simple change 😗

Do it for atleast testing purpose.

**Comment by yesahem** [comment]
> > > @pantharshit007 Go to neon.tech spin up a new postgres database, then afterwards go to prisma accelerate generate an accelerate url. Then create a .env in the backend folder with these three Variables.
> > > ```shell
> > > DIRECT_DATABASE_URL=Postgres db url of neon
> > > 
> > > DATABASE_URL=accelerate url
> > > 
> > > JWT_SECRET=your secret
> > > ```
> > 
> > 
> > I know all of that it's just that I don't wana do all that for a simple change 😗
> 
> Do it for atleast testing purpose.

for local testing since you are using a direct url you doesn't need a accurate 
BTW we don't need this feature as of now !!!
Thanks for your contribution !

**Pull request closed without merging**